### PR TITLE
Initial commit of context manager to spawn/destroy clusters

### PR DESCRIPTION
**Pull request by vnlitvinov**
## What do these changes do?
Add basic interface for controlling a cluster from within Modin.
Should be used for running distributed workloads with simply wrapping the workload in 
```python
with modin.experimental.cloud.Cluster(...):
    # do stuff
```

- [ ] passes `flake8 modin`
- [ ] passes `black --check modin`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #? 
- [ ] tests added and passing

### CODE REVIEW

**Comment by vnlitvinov** [comment]
cc @aregm @anmyachev

File: modin/experimental/cloud/cluster.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+class Cluster:
+    '''
+    Cluster manager for Modin. Knows how to use certain tools to spawn and destroy clusters,
+    can serve as context manager to keep cluster running only as long as it is needed.
+    '''
+
+    USER_NAME = 'modin'
+    IMAGE_NAME = 'moding-image'
+    KEY_NAME = 'modin-key'
+    DEFAULT_CLUSTER_NAME = 'modin-cluster'
```
Comment (anmyachev) [inline modin/experimental/cloud/cluster.py:89]: Maybe add `DEFAULT_WORKDIR = '~/.modin/cloud'` ?

File: modin/experimental/cloud/cluster.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        '''
+
+        # TODO: implement by reading 'Rhoc state'
+        raise NotImplementedError()
+
+    def __enter__(self):
+        self.spawn()
+        return self
+
+    def __exit__(self, *a, **kw):
```
Comment (anmyachev) [inline modin/experimental/cloud/cluster.py:172]:
```suggestion
    def __exit__(self, *args, **kwargs):
```

File: modin/experimental/cloud/cluster.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+            'key_name': self.KEY_NAME
+        }
+        if self.project_name:
+            params['project_name'] = self.project_name
+        if self.instance_type_worker_node:
+            params['instance_type_worker_node'] = self.instance_type_worker_node
+        
+        joint = ','.join(f'{param_name}={param_value}' for (param_name, param_value) in params.items())
+
+        return self.provider._make_rhoc_params() + ['--vars', joint]
```
Comment (anmyachev) [inline modin/experimental/cloud/cluster.py:133]: `_make_rhoc_params` undefined

**Pull request closed without merging**